addons/sale_purchase_project/models/purchase_order.py

```diff
@@ -0,0 +1,34 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    reinvoiced_so_count = fields.Integer(compute='_compute_reinvoiced_so_count')
+
+    def _compute_reinvoiced_so_count(self):
+        for order in self:
+            order.reinvoiced_so_count = len(order.invoice_ids.invoice_line_ids.reinvoiced_sale_line_id.order_id)
```

Comment:
Maybe try to use _read_group on indexed ``reinvoiced_sale_line_id`` if possible ?

Currently it's only displayed on form view but meh ^^